src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -285,6 +284,7 @@ private function createFirewalls(array $config, ContainerBuilder $container)
                 ->replaceArgument(1, $exceptionListener)
                 ->replaceArgument(2, $logoutListener)
                 ->replaceArgument(3, new Reference($configId))
+                ->addTag('security.firewall_map_context')
```

Comment:
Since #37368 has been merged, there now is a `security.firewalls` parameter containing all firewall names. I propose to use that in the compiler pass instead of introducing a tag (see the referenced PR for a usage example).

Ryan tried to not introduce SecurityBundle "internal" tags, as this might confuse users. E.g. PHPstorm will auto complete this, so we should try to not clutter the autocompletion list with tags the user should never use. (in the end, we should maybe introduce the concept of internal tags?)

Reply:
Rebased and changed to use `security.firewalls` now.